app/controllers/GameController.java

```diff
@@ -2,95 +2,42 @@
 
 import play.mvc.Controller;
 import play.mvc.Result;
-import services.GameLoop;
-import services.WebInput;
-import ttt.game.Board;
-import ttt.game.GameConstructor;
-import ttt.game.GameEngine;
+import services.GameHelper;
 import views.html.board;
 
-import java.util.Arrays;
-import java.util.List;
 import java.util.Map;
 
 public class GameController extends Controller {
 
-    private GameEngine game;
-    private String gametype;
-    private GameLoop gameLoop;
-    private int boardSize;
+    private GameHelper gameHelper;
 
     public GameController() {
-        this.gametype = null;
+        this.gameHelper = new GameHelper();
     }
 
-    public Result showBoard() {
-        String stringBoard = getBoard();
-        List<List<String>> rows = gameLoop.getRows();
-        return ok(board.render(rows, stringBoard, game.currentMark().toString(), gametype, gameLoop.isFinished(), gameLoop.getStatus()));
+    public Result showBoard() throws Exception {
+        return ok(board.render(gameHelper.getPresenter()));
     }
 
     public Result newBoard() {
         Map<String, String[]> request = request().body().asFormUrlEncoded();
-        String size = request.get("size")[0];
-        setBoardSize(size);
+        Integer size = Integer.valueOf(request.get("size")[0]);
+        gameHelper.setBoardSize(size);
         return redirect("/");
     }
 
     public Result newGame() {
         Map<String, String[]> request = request().body().asFormUrlEncoded();
         Integer type = Integer.valueOf(request.get("type")[0]);
         String gameType = request.get("name")[0];
-        GameEngine game = createGame(type, gameType);
-        this.gameLoop = new GameLoop(game);
+        gameHelper.createGame(type, gameType);
         return redirect("/game");
     }
 
     public Result placeMark() {
         Map<String, String[]> request = request().body().asFormUrlEncoded();
-        Integer row = Integer.valueOf(request.get("rowNumber")[0]);
-        Integer cell = Integer.valueOf(request.get("cellPosition")[0]);
-        playGame(row, cell);
+        Integer move = Integer.valueOf(request.get("position")[0]);
+        gameHelper.playGame(move);
         return redirect("/game");
     }
```

Comment:
GameController now only deals with receiving requests and dealing with those requests.